### PR TITLE
Upgrade Laravel and PHP to latest

### LINKED ISSUE #50: Upgrade Laravel
8 is getting long in the tooth and may be an issue for upgrading PHP.

### PR DESCRIPTION
**Pull request by qskousen**
Upgrade Laravel from 9 to 11.
Upgrade PHP from 8.1 to 8.3.
Get rid of most uses of `$_ENV`.
Various cleanup and fixes.

Closes #50, #51

### CODE REVIEW

**Comment by qskousen** [comment]
The pipeline is broken, doesn't like installing the specific npm version.

File: app/Http/Kernel.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Kernel extends HttpKernel
     protected $middleware = [
         // \App\Http\Middleware\TrustHosts::class,
         \App\Http\Middleware\TrustProxies::class,
-        \Fruitcake\Cors\HandleCors::class,
+        \Illuminate\Http\Middleware\HandleCors::class,
```
Comment (qskousen) [inline app/Http/Kernel.php:19]: This is now part of Laravel

File: app/Http/Middleware/TrustProxies.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace App\Http\Middleware;
 
-use Fideloper\Proxy\TrustProxies as Middleware;
+use Illuminate\Http\Middleware\TrustProxies as Middleware;
```
Comment (qskousen) [inline app/Http/Middleware/TrustProxies.php:5]: This is now part of Laravel

File: composer.json [hunk 3]
```diff
@@ -8,40 +8,37 @@
...
-        "php": "^8.1",
-        "alexpechkarev/google-maps": "^9.0",
+        "php": "^8.3",
+        "alexpechkarev/google-maps": "^10.0",
         "aws/aws-sdk-php": "~3.0",
         "barryvdh/laravel-debugbar": "^3.6",
-        "barryvdh/laravel-dompdf": "^1.0",
+        "barryvdh/laravel-dompdf": "^2.2",
         "barryvdh/laravel-snappy": "^1.0",
-        "doctrine/dbal": "^3.1",
```
Comment (qskousen) [inline composer.json:17]: Doctrine is no longer required

File: config/sanctum.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<?php
```
Comment (qskousen) [inline config/sanctum.php:1]: This is the default Sanctum config.

File: database/migrations/2019_12_14_000001_create_personal_access_tokens_table.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<?php
```
Comment (qskousen) [inline database/migrations/2019_12_14_000001_create_personal_access_tokens_table.php:1]: Sanctum migrations are no longer auto-migrated, so the migration needed to be published.

File: database/migrations/2021_04_25_230328_create_event_distance_for_members.php [hunk 6]
```diff
@@ -17,8 +17,8 @@ public function up()
             $table->id();
             $table->unsignedBigInteger('event_id')->default(0);
             $table->unsignedBigInteger('user_id')->default(0);
-            $table->unsignedDecimal('miles',8,2)->default(0);
-            $table->unsignedDecimal('minutes',8,2)->default(0);
+            $table->decimal('miles',8,2)->default(0);
```
Comment (qskousen) [inline database/migrations/2021_04_25_230328_create_event_distance_for_members.php:20]: Unsigned decimal is no longer supported, and in fact was never finalized in MySQL.

File: database/migrations/2021_05_15_084926_make_tags_nullable_on_colorways.php [hunk 7]
```diff
@@ -15,9 +15,9 @@ public function up()
     {
         Schema::table('colorways', function (Blueprint $table) {
             //
-            $table->text('colorway_description')->nullable()->default('')->change();
```
Comment (qskousen) [inline database/migrations/2021_05_15_084926_make_tags_nullable_on_colorways.php:18]: I am not sure why but MySQL didn't like specifying a default on TEXT.

File: docker-compose.yml [hunk 8]
```diff
@@ -47,6 +47,8 @@ services:
           test: ["CMD", "mysqladmin", "ping"]
     mysql-test:
         image: 'mysql:8.1'
+        ports:
```
Comment (qskousen) [inline docker-compose.yml:50]: I had to add this for my Podman. Shouldn't break anything on Docker though.

File: docker/app.dockerfile [hunk 9]
```diff
@@ -1,13 +1,22 @@
-FROM php:8.1-fpm AS php-build
+FROM php:8.3-fpm AS php-build
 
 RUN apt-get update && apt-get install -y  \
-    libzip-dev \
+    libzip-dev git libmagickwand-dev libzip-dev \
     --no-install-recommends \
-    && docker-php-ext-install pdo_mysql zip exif
+    && docker-php-ext-install pdo_mysql zip exif && \
```
Comment (qskousen) [inline docker/app.dockerfile:6]: PHP 8.3 Imagick PECL install is currently broken. It has to be built from source. I am not sure if I did this in the right layer, and I am not sure what Imagick is used for, so I am not sure how to test if it still works.

File: package.json [hunk 10]
```diff
@@ -63,6 +63,7 @@
         "vue3-datepicker": "^0.2.4",
         "vue3-dropzone": "^0.0.7",
         "vue3-timepicker": "*",
-        "vuex": "^4.0.2"
+        "vuex": "^4.0.2",
+        "ziggy-js": "^2.3.0"
```
Comment (qskousen) [inline package.json:67]: Ziggy now has a JS package to to what the old import from `../vendor/ziggy/.....` used to do.

File: phpunit.xml [hunk 11]
```diff
@@ -1,34 +1,32 @@
...
-        <server name="APP_ENV" value="testing"/>
-        <server name="BCRYPT_ROUNDS" value="4"/>
-        <server name="CACHE_DRIVER" value="array"/>
-        <server name="DB_CONNECTION" value="mysql-test"/>
-        <server name="MAIL_MAILER" value="log"/>
-        <server name="QUEUE_CONNECTION" value="sync"/>
-        <server name="SESSION_DRIVER" value="array"/>
-        <server name="TELESCOPE_ENABLED" value="false"/>
-    </php>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/11.3/phpunit.xsd" bootstrap="vendor/autoload.php" colors="true" cacheDirectory=".phpunit.cache">
```
Comment (qskousen) [inline phpunit.xml:2]: After updating PHPUnit, it wanted me to migrate the XML file. This is the result of that.

**Comment by qskousen** [comment]
I should also note that I tested some things manually, but I was unable to complete a proposal because my PandaDoc trial has expired :sweat_smile:

File: app/Http/Controllers/ProposalsController.php [hunk 12]
```diff
@@ -378,7 +379,8 @@ public function sendContract(Request $request, Proposals $proposal)
             return redirect()->route('proposals')->withErrors('Proposal has not been approved. Cannot send out.');
         }
 
-        $this->make_pandadoc_contract($proposal);
+        $proposalService = new ProposalServices($proposal);
```
Comment (qskousen) [inline app/Http/Controllers/ProposalsController.php:382]: Looks like this was a miss from changing it in the function above.

File: database/migrations/2021_04_25_230328_create_event_distance_for_members.php (hunk 6 above)
Comment (eddimull) [reply database/migrations/2021_04_25_230328_create_event_distance_for_members.php:20]: Fortunately this feature was never finished. So it'll be shcrödinger's feature.

File: docker/app.dockerfile (hunk 9 above)
Comment (eddimull) [reply docker/app.dockerfile:6]: It's always Imagick and/or gd. 

I think this was used for the image preview on the advance. I am not 100% positive though.

File: package.json (hunk 10 above)
Comment (eddimull) [reply package.json:67]: There's been a couple updates to ziggy as inertia updated. I might have missed this one.

**Review by eddimull — APPROVED** [review]

**Pull request merged**